pinot-core/src/test/java/com/linkedin/pinot/core/segment/index/creator/SegmentPartitionTest.java

```diff
@@ -178,6 +178,62 @@ public void testPruner() {
   }
 
   /**
+   * Unit test for {@link PartitionSegmentPruner}.
+   * <ul>
+   *   <li> Generates queries with in predicate on partitioned column with random values. </li>
+   *   <li> Ensures that if any column values that are in partition range ([0 5]) do not prune the segment,
+   *        whereas other values do. </li>
+   *   <li> Ensures that predicates on non-partitioned columns do not prune the segment. </li>
+   * </ul>
+   */
+  @Test
+  public void testPrunerIn() {
```

Comment:
What happens when we have more predicates? We wouldn't want to copy-paste-slightly-modify for which of those predicates, right? We should refactor the code so that common parts can be reused, and only test specific code is written within the individual test.